### DISCUSSION TITLE
How to specify units (e.g. mm or inches)?

### DISCUSSION THREAD

**Discussion by xpe** [hannobraun/fornjot#2193] [Q&A] [answered]
I'm looking at the [cuboid example in the models directory](https://github.com/hannobraun/fornjot/blob/main/models/cuboid/src/lib.rs). I'm not (yet) seeing where units are specified (or implied). How does one do this?

**Comment by hannobraun** [comment, marked as answer]
One doesn't :smile: 

It's a good question, but it's just not something I have looked into so far. The only place (as far as I remember) where units play a role, is exporting to 3MF (as STL doesn't support specifying a unit; again, as far as I remember). [3mf-rs](https://crates.io/crates/threemf) supports specifying the unit, but Fornjot simply doesn't, so it [defaults to millimeters](https://github.com/hannobraun/3mf-rs/blob/6c3192eb78992bf4b5719db6cfb9f112f7f58af2/src/model/mod.rs#L104-L108).

It should be pretty straight-forward to expose unit selection via API and/or CLI argument, when exporting to 3MF. If that's something you need, please let me know and I'd be happy to look into it (or merge a pull request that does it).

Long-term, it probably makes sense to handle units as part of the core kernel code, not just exporting. But that's something I'd like to defer for the foreseeable future, unless there's a real need. There are bigger problems right now.

**Reply by xpe** [reply]
Millimeters are the way to go for sure! (I'm one of the many Americans who think Imperial units are quite silly.)
